apollo-router/src/plugins/telemetry/config_new/events.rs

```diff
@@ -0,0 +1,66 @@
+use schemars::JsonSchema;
+use serde::Deserialize;
+
+use crate::plugins::telemetry::config_new::attributes::Extendable;
+use crate::plugins::telemetry::config_new::attributes::RouterAttributes;
+use crate::plugins::telemetry::config_new::attributes::RouterCustomAttribute;
+use crate::plugins::telemetry::config_new::attributes::SubgraphAttributes;
+use crate::plugins::telemetry::config_new::attributes::SubgraphCustomAttribute;
+use crate::plugins::telemetry::config_new::attributes::SupergraphAttributes;
+use crate::plugins::telemetry::config_new::attributes::SupergraphCustomAttribute;
+
+/// Events are
+#[allow(dead_code)]
+#[derive(Deserialize, JsonSchema, Clone, Default, Debug)]
+#[serde(deny_unknown_fields, default)]
+pub(crate) struct Events {
+    /// Router service events
+    router: Extendable<RouterEvents, Event<RouterAttributes, RouterCustomAttribute>>,
+    /// Subgraph service events
+    supergraph: Extendable<RouterEvents, Event<SupergraphAttributes, SupergraphCustomAttribute>>,
+    /// Supergraph service events
+    subgraph: Extendable<RouterEvents, Event<SubgraphAttributes, SubgraphCustomAttribute>>,
+}
+
+#[allow(dead_code)]
+#[derive(Clone, Deserialize, JsonSchema, Debug, Default)]
+#[serde(deny_unknown_fields, default)]
+struct RouterEvents {}
+
+#[allow(dead_code)]
+#[derive(Clone, Deserialize, JsonSchema, Debug, Default)]
+#[serde(deny_unknown_fields, default)]
+struct SupergraphEvents {}
+
+#[allow(dead_code)]
+#[derive(Clone, Deserialize, JsonSchema, Debug, Default)]
+#[serde(deny_unknown_fields, default)]
+struct SubgraphEvents {}
```

Comment:
Do you have example of fields we might have here ?

Reply:
[7c3414c](https://github.com/apollographql/router/pull/4061/commits/7c3414c45c8c4e49a6e26c607865e19ab1a227fe)